nda-transform/count_assoc_files.py

```diff
@@ -1,81 +1,138 @@
+"""
+Counts unique and duplicate files for NDA submissions
+"""
 
+import argparse
 from pathlib import Path
-import pandas as pd
-import sys
-
-# Usage:
-# at AWS workspace:
-# __file__ 65021 3705
-# at ERIS cluster:
-# __file__ 0 0
-
-nda_submissions_path = Path("/volumes/prod-ampscz/nda-submissions")
-id = sys.argv[1]
-collection = sys.argv[2]
-
-if id=='0' or collection=='0':
-    source_path = Path('/data/predict1/to_nda/nda-submissions/network_combined')
-    files = source_path.glob(f'*csv')
-else:
-    source_path = Path(f"/volumes/prod-ampscz/collaboration-space/{collection}")
-    files = source_path.glob(f"*/csv/{id}/part-*csv")
-
-file_columns='data_file1 data_file2 image_file bvalfile bvecfile transcript_file'.split()
-
-count=0
-unique_file={}
-duplicate={}
-for file in files:
-    print("Checking", file)
-
-    
-    if id=='0' or collection=='0':
-        _data_file_path = source_path
-        df = pd.read_csv(file, header=1)
-    else:
-        df = pd.read_csv(file)
-        _data_file_path = nda_submissions_path / df.loc[0]["SUBMISSION_FOLDER_NAME"]
+from typing import Dict, List, Literal, Set, Optional
 
-
-    # check a file only if at least one file_columns exist
-    exist=0
-    for c in file_columns:
-        if c in df.columns:
-            exist=1
-            break
-    if not exist:
-        continue
+import pandas as pd
 
 
-    for idx, row in df.iterrows():
-        
+def count_files(
+    project_path: Optional[Path],
+    source_path: Path,
+    files: List[Path],
+    context: Literal["aws", "eris"],
+):
+    """
+    Count unique and duplicate files in the given list of files
+    """
+    files_count: int = 0
+    unique_files: Set = set()
+    duplicate_files_count: Dict[str, int] = {}
+
+    file_columns = (
+        "data_file1 data_file2 image_file bvalfile bvecfile transcript_file".split()
+    )
+
+    for file in files:
+        print(f"Checking {file}", end="")
+
+        if context == "eris":
+            _data_file_path = source_path
+            df = pd.read_csv(file, header=1)
+        else:
+            nda_submissions_path = project_path / "nda-submissions"
+            df = pd.read_csv(file)
+            _data_file_path = nda_submissions_path / df.loc[0]["SUBMISSION_FOLDER_NAME"]
+
+        # check a file only if at least one file_columns exist
+        check_file = False
         for c in file_columns:
-        
             if c in df.columns:
-                if not pd.isna(row[c]):
-                    data_file = row[c]
-
-                    data_file_path: Path = _data_file_path / data_file
-
-                    if not data_file_path.exists():
-                        print(f"File does not exist: {data_file_path}")
-                    else:
-                        count+=1
-
-                        if data_file_path not in unique_file:
-                            unique_file[data_file_path]=1
+                check_file = True
+                break
+
+        if not check_file:
```

Comment:
Dheshan, please remove this block and reinstate mine. it is too verbose as most files do not have them.

Reply:
You want me to replace my block:

```python
        # check a file only if at least one file_columns exist
        check_file = False
        for c in file_columns:
            if c in df.columns:
                check_file = True
                break
```

with this block from yours:
```python
    # check a file only if at least one file_columns exist
    exist=0
    for c in file_columns:
        if c in df.columns:
            exist=1
            break
    if not exist:
        continue
```

Reply:
No. Remove lines 48-52.

Reply:
Should I then also remove L31:

```python
print(f"Checking {file}", end="")
```

Reply:
Why is this `end=""` necessary? Why was my message not sufficient?